website/src/_posts/2022-12-3.3.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "🎄 Uppy 3.1—3.3: Improved AWS S3 Multipart, Single File Mode" 
+date: 2022-12-23
+author: 
+  - aduh95
+  - arturi
+  - murderlon
+  - mifi
+image: "https://uppy.io/images/blog/3.1-3.3/single-file-mode.jpg"
+published: true
+---
+
+Three minor releases ago we’ve released Uppy 3.0. It’s time to give you an update on what we’ve been working on all this time.
+
+TL;DR: AWS S3 Multipart stability improvements, Single File Mode, more tests and bugfixes, new locales.
+
+## AWS S3 Multipart: improved performance on slow connections
+
+We’ve been doing a lot of tests to find out why some uploads with `@uppy/aws-s3-multipart` plugin failed. It turned out, simultaneously signing and uploading many chunks at once saturated the HTTP/1.1 connection limit, which lead to signatures expiring by the time Uppy started uploading the next chunk.
+
+We’ve refactored our internal queue mechanism to sign chunks right before we start uploading them to the remote server. When the upload is paused, the queue is emptied and re-populated again on resume. The default `limit` for this plugin has been set to `6`. See [PR #4187](https://github.com/transloadit/uppy/pull/4187) for more details.
+
+To improve things even further, the default S3 expiry has been lifted from `300` to `800` seconds ([PR #4206](https://github.com/transloadit/uppy/pull/4206)).
+
+@uppy/aws-s3-multipart: handle slow connections better (Antoine du Hamel / #4213)
```

Comment:
- "uploads failed" is a bit generic and doesn't put the plugin in a good spotlight
- some context is missing
- "improve things even further" and then talk about simply upping the expiry might sounds like a 'hack and slash' solution to people if we don't provide context
- I don't think it's necessary to link PRs

```suggestion
`@uppy/aws-s3-multipart` has been serving well for people who prefer a _client-to_s3_ setup with Uppy. However, some uploads could still fail in poor network conditions. Since uploads to S3 always use a HTTP/1.1 connection, which means there is a limit to concurrent connections, we found out that it was possible for an upload to expire on poor connections before it was fully uploaded.

We have have now redone our [rate limiting](https://en.wikipedia.org/wiki/Rate_limiting) specifically for this plugin to make sure no more connections are created than it can handle in poor conditions.
```